---
title: "recount_brain cross-study example"
author:
- name: Shannon E. Ellis
  affiliation: Johns Hopkins Bloomberg School of Public Health
  email: sellis18@jhmi.edu
output:
  BiocStyle::html_document:
      code_folding: hide
abstract: |
  Example on how to use recount_brain metadata across multiple studies. We show how to download data from recount2, add the sample metadata from recount_brain, explore the sample metadata and the gene expression data, and perform a cross-study analysis.
editor_options: 
  chunk_output_type: inline
---

```{r 'setup', echo = FALSE, warning = FALSE, message = FALSE}
## Bib setup
library('knitcitations')
library('BiocStyle')

## Load knitcitations with a clean bibliography
cleanbib()
cite_options(hyperlink = 'to.doc', citation_format = 'text', style = 'html')

## Write bibliography information
bib <- c(
    R = citation(),
   # BiocStyle = citation('BiocStyle'),
    devtools = citation('devtools'),
    knitcitations = citation('knitcitations'),
    knitr = citation('knitr')[3],
	  recount = citation('recount')[1],
    recountWorkflow = citation('recount')[2],
    rmarkdown = citation('rmarkdown'),
    original = bib_metadata('10.1101/gr.165126.113'),
    limma = citation('limma'),
    voom = RefManageR::BibEntry('article', key = 'voom', author = 'CW Law and Y Chen and W Shi and GK Smyth', year = '2014', title = 'Voom: precision weights unlock linear model analysis tools for RNA-seq read counts', journal = 'Genome Biology', volume = '15', pages = 'R29')
)

write.bibtex(bib, file = 'example_multistudy.bib')

## Set colors
## import colors to use
  bright= c(red=rgb(222,45,38, maxColorValue=255), #de2d26
            pink=rgb( 255, 102, 153, maxColorValue=255), #ff6699
            orange=rgb(232,121,12, maxColorValue=255),   #e8790c
            yellow=rgb(255,222,13, maxColorValue=255), #ffde0d          
            green=rgb(12,189,24, maxColorValue=255),  #0cbd18           
            teal=rgb(59,196,199, maxColorValue=255), #3bc4c7
            blue=rgb(58,158,234, maxColorValue=255), #3a9eea
            purple=rgb(148,12,232, maxColorValue=255)) #940ce8  
```

<a href="https://jhubiostatistics.shinyapps.io/recount/"><img src="https://raw.githubusercontent.com/LieberInstitute/recount-brain/master/recount_brain.png" align="center"></a>

# Introduction

This document is an example of how you can use `recount_brain`. We will first explore what samples are available in `recount_brain`. After determining what samples are of most interest to us, we will download the expression data for the studies from which these samples were generated. We will remove batch effects across studies. Finally, we will show how to add the `recount_brain` metadata and perform a gene differential expression analysis using this information to assess replicability across studies. Specifically, we'll be looking to see if the same genes are most variable across different glioblastoma datasets. We'll use a kidney cancer data set as our negative control data set. We'll use concordance across varible genes between datasets as our measure of interest.

## Load libraries

Since we will be using many functions from the `recount` package, let's load it first^[If you are a first time `recount` user, we recommend first reading the package vignette at [bioconductor.org/packages/recount](http://bioconductor.org/packages/recount).]. We'll also download all of the other 

```{r 'load-packages', echo=FALSE, warning = FALSE, message = FALSE}
library(recount) # version >1.5.9
library(dplyr)
library(downloader)
library(janitor)
# install_github('LieberInstitute/jaffelab')
library(jaffelab)
library(ggplot2)
library(sva)
```

# The Data

## Sample metadata

To explore the samples available, we'll first search among TCGA samples and available samples in `recount-brain` to determine which samples and which studies we'll use for our analysis. We'll similarly load the TCGA meta data. We will use samples from TCGA and from `recount-brain` in our analysis.

```{r 'metadata', message = FALSE}
## get recount_brain metadata
recount_brain_v1 <- recount::add_metadata(source = 'recount_brain_v1')
brain = recount_brain_v1

## get TCGA metadata
tcga <- as.data.frame(recount::all_metadata('TCGA'))
```

## Identifying studies 

Among the TCGA samples, there are 707 samples with either Glioma or Glioblastoma Multiforme (GBM). There are four `recount_brain` studies with at least 20 cancer samples. Of these four studies, three studies have Glioblastoma samples. Two of these three studies have more than 20 samples. We'll use these two studies (`SRP027383` and `SRP044668`) to look for variability in expression across studies. 

## TCGA brain cancer samples
```{r}
# Find TCGA samples that are brain cancer samples
tcga %>% 
  filter(gdc_cases.project.primary_site == "Brain") %>%
  group_by(gdc_cases.project.name) %>% 
  summarise (n = n())
```

## recount-brain cancer studies
```{r}
# Find recount-brain studies that are cancer studies
(studies <- brain %>% filter(!is.na(tumor_type)) %>%   
  group_by(sra_study_s) %>% 
  summarise (n = n()) %>% 
  filter(n>=20))
```

## Cancer type overlap (TCGA, recount-brain)
```{r}
# Find recount-brain studies with cancer type that overlaps with TCGA
(GBM_samples <- brain %>% 
  filter(sra_study_s %in% studies$sra_study_s) %>% 
  group_by(sra_study_s, tumor_type) %>% 
  summarise (n = n()) %>% 
  filter(tumor_type == "Glioblastoma") %>% 
  filter(n>=20))
```

## Download expression data

Just like any study in `recount2` `r citep(bib[['recount']])`, we first need to download the gene count data using `recount::download_study()`. We'll additionally use the `add_metadata()` function to include the `recount-brain` metadata information for these two studies. Addiitonally, below, we read in the TCGA gene expression data from brain samples.

```{r 'getgene', message=FALSE}
# SRA (recount-brain) expression data
todownload <- GBM_samples$sra_study_s

for(i in 1:length(todownload)){
  if(!file.exists(file.path(todownload[i], 'rse_gene.Rdata'))) {
      download_study(todownload[i])
  } else {
      load(file.path(todownload[i], 'rse_gene.Rdata'), verbose = TRUE)
      assign(paste0("rse_gene_", todownload[i]), rse_gene)
  }
}

# add recount-brain metadata
rse_gene_SRP027383 <- recount::add_metadata(rse_gene_SRP027383) 
rse_gene_SRP044668 <- recount::add_metadata(rse_gene_SRP044668) 

# TCGA brain expression data 
if(!file.exists(file.path('TCGA', 'rse_gene_brain.Rdata'))) {
  downloader::download('http://duffel.rail.bio/recount/v2/TCGA/rse_gene_brain.Rdata', destfile = 
                'rse_gene_brain.Rdata', mode = 'wb')
} else {
  load(file.path('TCGA','rse_gene_brain.Rdata'))
  assign('rse_gene_TCGA', rse_gene)
}

```

## Expresion Counts

After loading the expression data from our studies of interest, we filter to only include TCGA disease samples, requiring samples to have Glioblastoma (GBM) and to be samples from the primary tumor, rather than nearby healthy tissue or more advanced tissue sample (i.e. a metastasis). We combine all our data into a single object, so that when we scale the data in the next steps, it can all be completed together.

```{r 'expression-counts', message=FALSE}
# get expression counts
## combine rses to scale counts all together
rse_gene_SRA <- cbind(rse_gene_SRP027383,rse_gene_SRP044668)

## just 15 Primary Tumor GBM samples in TCGA
tokeep <- (colData(rse_gene_TCGA)$gdc_cases.project.name=="Glioblastoma Multiforme" & colData(rse_gene_TCGA)$cgc_sample_sample_type=="Primary Tumor")
rse_gene_TCGA<-rse_gene_TCGA[,tokeep]

# get metadata 
# add dataset column for easy tracking later
tcga_md <-  as.data.frame(colData(rse_gene_TCGA)) %>% 
  mutate(dataset='TCGA', disease_status='Disease', tumor_type="Glioblastoma") %>%
  filter(cgc_sample_sample_type=="Primary Tumor")
SRP027383_md <- as.data.frame(colData(rse_gene_SRP027383)) %>% 
  mutate(dataset='SRP027383')
SRP044668_md <- as.data.frame(colData(rse_gene_SRP044668)) %>% 
  mutate(dataset='SRP044668')

## make sure that same metadata columns are present so that RSEs can be merged
cols_to_bind <- colnames(colData(rse_gene_TCGA))[colnames(colData(rse_gene_TCGA)) %in% colnames(colData(rse_gene_SRA))]
colData(rse_gene_SRA)  <- colData(rse_gene_SRA)[,cols_to_bind]
colData(rse_gene_TCGA)  <- colData(rse_gene_TCGA)[,cols_to_bind]

## merge data so that it can all be scaled together
rse_gene_total <- cbind(rse_gene_TCGA, rse_gene_SRA)


# Expression and metadata combined across data sets
# 538 samples
md <- bind_rows(tcga_md, SRP027383_md) %>%
        bind_rows(., SRP044668_md)
```

# Quality Control

There are a number of critical quality control steps that are necessary before we can make any cross-study comparisons. We'll (1) filter out lowly-expressed genes, (2) remove outlier samples, and (3) normalize the data to remove unwanted sources of variation from the data. 

## Filter genes (low expression)

```{r 'exprcut', fig.cap = c('Number of genes expressed at given mean RPKM cutoff.', 'Percent of genes epxressed at a given mean RPKM cutoff.', 'Distribution of number of expressed samples across all genes at a given mean RPKM cutoff')}
## remove lowly expressed genes
rse_rpkm <- getRPKM(scale_counts(rse_gene_total)) # 707 samples 

## Compute RPKM and mean RPKM
rpkm_mean <- rowMeans(rse_rpkm)
## Esmate a mean RPKM cutoff
expr_cuts <- expression_cutoff(rse_rpkm)
#round(mean(expr_cuts), 2)

## Filter genes with low levels of expression
rpkm <- rse_rpkm[rpkm_mean > round(mean(expr_cuts), 2),]
rpkm_log2 <- log2(rpkm+0.5)
```
Here, we have scale the data and caclulate RPKM for all genes in the data set (N = `r nrow(rse_gene_total)`. We then remove lowly expressed genes, defining lowly expressed genes is a mean expression across samples < `r round(mean(expr_cuts), 2)`. This leaves `r nrow(rpkm_log2)` genes for downstream analysis. 

## Run PCA

Having filtered out lowly-expressed genes, we now run a principal component analysis (PCA) to look at global expression patterns across this data set. Below we see that, as expected, samples are clustering by study. This is something we will have to correct for before carrying out our analysis. 

```{r 'pca', cache=TRUE, warning=FALSE}
## Look at mean variance relationship
## Calculate PCs with svd function
expr.pca.rpkm.log2 <- svd(rpkm_log2 - rowMeans(rpkm_log2))

par(font.lab = 2, cex.lab = 1.2, font.axis = 2, cex.axis = 1.2)
plot(expr.pca.rpkm.log2$v[, 1], expr.pca.rpkm.log2$v[, 2], col= as.factor(md$dataset) , pch = 19, cex = 1.5,
     xlab = 'PC1', ylab = 'PC2',
     main = 'PC (gene level) : log2(RPKM + 0.5)')
legend('bottomright', pch = 19, col=c(1:3),
       names(summary(as.factor(md$dataset))))

## Are healthy sample clustered?
par(font.lab = 2, cex.lab = 1.2, font.axis = 2, cex.axis = 1.2)
cols <- c(bright[c(1)],"grey48")
plot(expr.pca.rpkm.log2$v[, 1], expr.pca.rpkm.log2$v[, 2], col= cols[as.factor(md$disease_status)] , pch = 19, cex = 1.5,
     xlab = 'PC1', ylab = 'PC2',
     main = 'PC (gene level) : log2(RPKM + 0.5) colored by Disease Status')
legend('bottomright', pch = 19, col=c(1:2),
       names(summary(as.factor(md$disease_status))))
#dev.off()
```


## Remove controls

Given the fact that control samples are clustering in the PCA plot above, we'll remove these from the analysis prior to normalizing the data to remove dataset specific effects. After removing these control samples from analysis, we see that the data still cluster by data set.


```{r data-pca, warning=FALSE, cache=TRUE, fig.height=8}
## remove 19 control samples and just include GBM samples
rpkm_log2 <- rpkm_log2[,(md$tumor_type=="Glioblastoma" & !is.na(md$tumor_type))]
md <- md[(md$tumor_type=="Glioblastoma" & !is.na(md$tumor_type)),]

tabyl(md$dataset)

## Run pca
expr.pca.counts.log2 <- svd(rpkm_log2 - rowMeans(rpkm_log2))

## visualize
par(mfrow=c(2,2))
par(font.lab = 2, cex.lab = 1.2, font.axis = 2, cex.axis = 1.2)
plot(expr.pca.counts.log2$v[, 1], expr.pca.counts.log2$v[, 2], col= as.factor(md$dataset) , pch = 19, cex = 1.5,
     xlab = 'PC1', ylab = 'PC2',
     main = 'PC (gene level) : log2(scaled counts + 0.5)')

par(font.lab = 2, cex.lab = 1.2, font.axis = 2, cex.axis = 1.2)
plot(expr.pca.counts.log2$v[, 3], expr.pca.counts.log2$v[, 4], col= as.factor(md$dataset) , pch = 19, cex = 1.5,
     xlab = 'PC3', ylab = 'PC4',
     main = 'PC (gene level) : log2(scaled counts + 0.5)')

par(font.lab = 2, cex.lab = 1.2, font.axis = 2, cex.axis = 1.2)
plot(expr.pca.counts.log2$v[, 5], expr.pca.counts.log2$v[, 6], col= as.factor(md$dataset) , pch = 19, cex = 1.5,
     xlab = 'PC5', ylab = 'PC6',
     main = 'PC (gene level) : log2(scaled counts + 0.5)')

par(font.lab = 2, cex.lab = 1.2, font.axis = 2, cex.axis = 1.2)
plot(expr.pca.counts.log2$v[, 7], expr.pca.counts.log2$v[, 8], col= as.factor(md$dataset) , pch = 19, cex = 1.5,
     xlab = 'PC7', ylab = 'PC8',
     main = 'PC (gene level) : log2(scaled counts + 0.5)')
legend('bottomright', pch = 19, col=c(1:3),
       names(summary(as.factor(md$dataset))))

#Plot Variance Explained
par(mfrow=c(1,1))
plot(expr.pca.counts.log2$d^2/sum(expr.pca.counts.log2$d^2), xlim = c(0, 15), type = "b", pch = 16, xlab = "principal components", 
    ylab = "variance explained")
```

## Normalize for Dataset

To remove effects of TCGA vs SRA data sets (as they separate out in initial PCA), we first adjust the expression data and remove the effects of dataset. As TCGA is separating out in PC1 above while the two SRA samples are not, we code dataset as 1 if TGCA and 0 otherwise. 

```{r data-normalization, warning=FALSE, cache=TRUE, fig.height=8}

## code dataset where SRA dataset is 0 and TCGA is 1
dataset <- md$dataset
dataset[md$dataset=="SRP027383"] <- 0
dataset[md$dataset=="SRP044668"] <- 0
dataset[md$dataset=="TCGA"] <- 1
dataset <- as.numeric(dataset)

## function to normalize expression removing effects of dataset
normalize <- function(x){
  #run model
  lmfit <- lm(x ~ dataset)
  #adjust data
  #note: for quantitative, you have to subtract person's age from the average age of the group; for binary traits, you don't have to worry about this
  out <- x -  (coef(lmfit)["dataset"]*(dataset)) 
  return(out)
}

df <- apply(rpkm_log2,1,normalize)
rpkm_normalized <- t(df)

expr.pca.normalized <- svd(rpkm_normalized - rowMeans(rpkm_normalized))

par(mfrow=c(2,2))
par(font.lab = 2, cex.lab = 1.2, font.axis = 2, cex.axis = 1.2)
plot(expr.pca.normalized$v[, 1], expr.pca.normalized$v[, 2], col= as.factor(md$dataset) , pch = 19, cex = 1.5,
     xlab = 'PC1', ylab = 'PC2',
     main = 'PC (gene level) : log2(normzlized counts + 0.5)')

par(font.lab = 2, cex.lab = 1.2, font.axis = 2, cex.axis = 1.2)
plot(expr.pca.normalized$v[, 3], expr.pca.normalized$v[, 4], col= as.factor(md$dataset) , pch = 19, cex = 1.5,
     xlab = 'PC3', ylab = 'PC4',
     main = 'PC (gene level) : log2(normzlized counts + 0.5)')


par(font.lab = 2, cex.lab = 1.2, font.axis = 2, cex.axis = 1.2)
plot(expr.pca.normalized$v[, 5], expr.pca.normalized$v[, 6], col= as.factor(md$dataset) , pch = 19, cex = 1.5,
     xlab = 'PC5', ylab = 'PC6',
     main = 'PC (gene level) : log2(normzlized counts + 0.5)')


par(font.lab = 2, cex.lab = 1.2, font.axis = 2, cex.axis = 1.2)
plot(expr.pca.normalized$v[, 7], expr.pca.normalized$v[, 8], col= as.factor(md$dataset) , pch = 19, cex = 1.5,
     xlab = 'PC7', ylab = 'PC8',
     main = 'PC (gene level) : log2(normzlized counts + 0.5)')
legend('bottomright', pch = 19, col=c(1:3),
       names(summary(as.factor(md$dataset))))

#Plot Variance Explained
par(mfrow=c(1,1))
plot(expr.pca.normalized$d^2/sum(expr.pca.normalized$d^2), xlim = c(0, 15), type = "b", pch = 16, xlab = "principal components", 
    ylab = "variance explained")
```


## Adjust for PCs

Above we see that removing the effects of dataset is helpful, however, the data do not completely overlap. To address this, we then remove the effects of the first 6 PCs from the expression data. After running PCA on these normalized expression data, we see that the effects of dataset have been removed and cross-study comparisons are now possible.

```{r pca-adjust, warning=FALSE, cache=TRUE, fig.height=8}
PC1 <- expr.pca.normalized$v[, 1]
PC2 <- expr.pca.normalized$v[, 2]
PC3 <- expr.pca.normalized$v[, 3]
PC4 <- expr.pca.normalized$v[, 4]
PC5 <- expr.pca.normalized$v[, 5]
PC6 <- expr.pca.normalized$v[, 6]

normalize_PC <- function(x){
  #run model
  lmfit <- lm(x ~  + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 )
  #adjust data
  #note: for quantitative, you have to subtract person's age from the average age of the group; for binary traits, you don't have to worry about this
  out <- x - (coef(lmfit)["PC1"]*(PC1-mean(PC1))) - (coef(lmfit)["PC2"]*(PC2-mean(PC2))) - (coef(lmfit)["PC3"]*(PC3-mean(PC3))) - (coef(lmfit)["PC4"]*(PC4-mean(PC4))) - (coef(lmfit)["PC5"]*(PC5-mean(PC5))) - (coef(lmfit)["PC6"]*(PC6-mean(PC6))) 
  return(out)
}

df <- apply(rpkm_normalized,1,normalize_PC)
rpkm_normalized_PC <- t(df)

expr.pca.normalized.pc <- svd(rpkm_normalized_PC - rowMeans(rpkm_normalized_PC))

par(mfrow=c(2,2))
par(font.lab = 2, cex.lab = 1.2, font.axis = 2, cex.axis = 1.2)
plot(expr.pca.normalized.pc$v[, 1], expr.pca.normalized.pc$v[, 2], col= as.factor(md$dataset) , pch = 19, cex = 1.5,
     xlab = 'PC1', ylab = 'PC2',
     main = 'PC (gene level) : log2(normzlized counts + 0.5)')

par(font.lab = 2, cex.lab = 1.2, font.axis = 2, cex.axis = 1.2)
plot(expr.pca.normalized.pc$v[, 3], expr.pca.normalized.pc$v[, 4], col= as.factor(md$dataset) , pch = 19, cex = 1.5,
     xlab = 'PC3', ylab = 'PC4',
     main = 'PC (gene level) : log2(normzlized counts + 0.5)')

par(font.lab = 2, cex.lab = 1.2, font.axis = 2, cex.axis = 1.2)
plot(expr.pca.normalized.pc$v[, 5], expr.pca.normalized.pc$v[, 6], col= as.factor(md$dataset) , pch = 19, cex = 1.5,
     xlab = 'PC5', ylab = 'PC6',
     main = 'PC (gene level) : log2(normzlized counts + 0.5)')

par(font.lab = 2, cex.lab = 1.2, font.axis = 2, cex.axis = 1.2)
plot(expr.pca.normalized.pc$v[, 7], expr.pca.normalized.pc$v[, 8], col= as.factor(md$dataset) , pch = 19, cex = 1.5,
     xlab = 'PC7', ylab = 'PC8',
     main = 'PC (gene level) : log2(normzlized counts + 0.5)')
legend('bottomright', pch = 19, col=c(1:3),
       names(summary(as.factor(md$dataset))))

```

# Glioblastoma Data Analysis

Having cleaned the data and made cross-study comparisons possible, we're now interested in answering our question of interest. Are the same genes most variable across different GBM studies? 


## Variable Expression Analysis

To first answer this question, we calculate variance for each gene within each study.

```{r variable-genes, warning=FALSE}
df = as.data.frame(t(rpkm_normalized_PC))

## expression from each dataset
## N=74
SRP044668_df <- df %>% 
  filter(md$dataset == 'SRP044668')

## N=175
SRP027383_df <- df %>% 
  filter(md$dataset == 'SRP027383')

## N=270
TCGA_df <- df %>% 
  filter(md$dataset == 'TCGA')

## is measure variance across each dataset?
SRP044668_vars <- colVars(as.matrix(SRP044668_df))
SRP027383_vars <- colVars(as.matrix(SRP027383_df))
TCGA_vars <- colVars(as.matrix(TCGA_df))

## add genenames back in
names(SRP044668_vars) <- names(SRP027383_vars) <- names(TCGA_vars) <- colnames(df)
```


## Concordance across studies

Having calculated within-study variance, we can then look at concordance at the top (CAT) plots to assess the results. To generate a CAT plot, we sort each study's genes by variance. Then, we compare the genes found in cross-study comparison to one another. If the analyses find the same genes, the line in the CAT plot will fall along the 45 degree (grey) line. The less concordant the results are, the furter from this 45 degree line, the results will fall.

```{r variable-concordance, cache=TRUE, warning=FALSE}
## sort by variance
p.mod1.sort <- SRP044668_vars[order(SRP044668_vars,decreasing=TRUE)]
p.mod2.sort <- SRP027383_vars[order(SRP027383_vars,decreasing=TRUE)]
p.mod3.sort <- TCGA_vars[order(TCGA_vars,decreasing=TRUE)]


conc <- NULL
conc_TCGA <- NULL
conc_TCGA2 <- NULL

for(i in 1:length(p.mod2.sort)){
    conc[i] <- sum(names(p.mod2.sort)[1:i] %in% names(p.mod1.sort)[1:i])
    conc_TCGA[i] <- sum(names(p.mod2.sort)[1:i] %in% names(p.mod3.sort)[1:i])
    conc_TCGA2[i] <- sum(names(p.mod1.sort)[1:i] %in% names(p.mod3.sort)[1:i])
}

## all genes
par(mfrow = c(1, 1), font.lab = 1.5, cex.lab = 1.2, font.axis = 1.5, cex.axis = 1.2)
plot(seq(1:length(p.mod2.sort)), conc, 
     type = 'l', las = 0,
     xlim = c(0, length(conc)),
     ylim = c(0, length(conc)),
     xlab = 'ordered genes in reference study',
     ylab = 'ordered genes in new study',
     main = 'Concordance')
for(k in 1:30){
    abline(v = k * 5000, cex = 0.5, col = 'lightgrey')
    abline(h = k * 5000, cex = 0.5, col = 'lightgrey')
}
abline(coef=c(0,1),col="grey48")
lines(seq(1:length(p.mod2.sort)), conc, col = bright[2], lwd = 3)
lines(seq(1:length(p.mod3.sort)), conc_TCGA, col = bright[5], lwd = 3)
lines(seq(1:length(p.mod3.sort)), conc_TCGA2, col = bright[8], lwd = 3)
legend('topleft', col = bright[c(2,5,8)], c("SRP044668_SRP027383",  "SRP027383_TCGA","SRP044668_TCGA"),lty=1,lwd=5, bg="white",bty="n")


## top 1000 genes
par(mfrow = c(1, 1), font.lab = 1.5, cex.lab = 1.2, font.axis = 1.5, cex.axis = 1.2)
plot(seq(1:1000), conc[1:1000], 
     type = 'l', las = 0,
     xlim = c(0, 1000),
     ylim = c(0, 1000),
     xlab = 'ordered genes in reference study',
     ylab = 'ordered genes in new study',
     main = 'Concordance')
for(k in 1:10){
    abline(v = k * 200, cex = 0.5, col = 'lightgrey')
    abline(h = k * 200, cex = 0.5, col = 'lightgrey')
}
abline(coef=c(0,1),col="grey48")
lines(seq(1:1000), conc[1:1000], col = bright[2], lwd = 3)
lines(seq(1:1000), conc_TCGA[1:1000], col = bright[5], lwd = 3)
lines(seq(1:1000), conc_TCGA2[1:1000], col = bright[8], lwd = 3)
legend('topleft', col = bright[c(2,5,8)], c("SRP027383_SRP044668", "SRP027383_TCGA","SRP044668_TCGA"),lty=1,lwd=5, bg="white",bty="n")

## most variable genes overlap across datasets?

```

Here, wee see that each cross-study comparsion shows a similar level of concordance between studies.

# Overlap with non-GBM samples

To compare these results to a non-GBM cancer study, we download TCGA kidney cancer data from `recount`. 

## Data

The data here are kindey primary tumor samples. We chose kidney because there are a relatively large number of available samples and because this tissue is biologically dissimilar from brain.

```{r non-GBM-data, cache=TRUE, warning=FALSE}

# take a look at what samples we have normal tissue for 
tcga %>% group_by(gdc_cases.project.primary_site,cgc_sample_sample_type) %>%
  summarise(n=n()) %>% 
  filter(cgc_sample_sample_type=="Solid Tissue Normal") %>% 
  arrange(-n)

# can see that there are a lot of kidney healthy and kidney tumor. Will compare these to GBM with hypothesis that GBM most similar to kidney and less similar to healthy but overall less similar to GBM comparison

if(!file.exists(file.path('TCGA', 'rse_gene_kidney.Rdata'))) {
  downloader::download('http://duffel.rail.bio/recount/v2/TCGA/rse_gene_kidney.Rdata', destfile = 
                'TCGA/rse_gene_kidney.Rdata', mode = 'wb')
  load(file.path('TCGA','rse_gene_kidney.Rdata'))
  assign('rse_gene_TCGA_kidney', rse_gene)
} else {
  load(file.path('TCGA','rse_gene_kidney.Rdata'))
  assign('rse_gene_TCGA_kidney', rse_gene)
}

rse_gene_TCGA_kidney <- rse_gene_TCGA_kidney[,(colData(rse_gene_TCGA_kidney)$cgc_sample_sample_type=="Primary Tumor")]

load(file.path('TCGA','rse_gene_brain.Rdata'))
md_t <- colData(rse_gene)

rse_combine_TCGA <- rse_gene_TCGA
tokeep2 <- (md_t$gdc_cases.project.name=="Glioblastoma Multiforme" & md_t$cgc_sample_sample_type=="Primary Tumor")
md_t<-md_t[tokeep,]
colData(rse_combine_TCGA) <- md_t

rse_TCGA <- cbind(rse_combine_TCGA, rse_gene_TCGA_kidney)
md_TCGA <- colData(rse_TCGA)
```

## Filter genes (low expression)

We again filter out lowly-expressed genes. Here we include the same genes for analysis as were used in the analysis above. 

```{r 'exprcut-tcga', fig.cap = c('Number of genes expressed at given mean RPKM cutoff.', 'Percent of genes epxressed at a given mean RPKM cutoff.', 'Distribution of number of expressed samples across all genes at a given mean RPKM cutoff')}
## remove lowly expressed genes
rse_rpkm_tcga <- getRPKM(scale_counts(rse_TCGA)) 

## Filter genes with low levels of expression (same genes as initial analysis)
rpkm_tcga <- rse_rpkm_tcga[rpkm_mean > round(mean(expr_cuts), 2),]
rpkm_log2_tcga <- log2(rpkm_tcga+0.5)
```

## Run PCA

We again assess global gene expression patterns using PCA. We see that samples are clustering by tissue, as expected.

```{r 'pca-tcga', cache=TRUE, warning=FALSE}
## Look at mean variance relationship
## Calculate PCs with svd function
expr.pca.rpkm.log2.tcga <- svd(rpkm_log2_tcga - rowMeans(rpkm_log2_tcga))

par(mfrow=c(2,2))
par(font.lab = 2, cex.lab = 1.2, font.axis = 2, cex.axis = 1.2)
plot(expr.pca.rpkm.log2.tcga$v[, 1], expr.pca.rpkm.log2.tcga$v[, 2], col= as.factor(md_TCGA$gdc_cases.project.primary_site) , pch = 19, cex = 1.5,
     xlab = 'PC1', ylab = 'PC2',
     main = 'PC (gene level) : log2(RPKM + 0.5)')

par(font.lab = 2, cex.lab = 1.2, font.axis = 2, cex.axis = 1.2)
plot(expr.pca.rpkm.log2.tcga$v[, 3], expr.pca.rpkm.log2.tcga$v[, 4], col= as.factor(md_TCGA$gdc_cases.project.primary_site) , pch = 19, cex = 1.5,
     xlab = 'PC3', ylab = 'PC4',
     main = 'PC (gene level) : log2(RPKM + 0.5)')

par(font.lab = 2, cex.lab = 1.2, font.axis = 2, cex.axis = 1.2)
plot(expr.pca.rpkm.log2.tcga$v[, 5], expr.pca.rpkm.log2.tcga$v[, 6], col= as.factor(md_TCGA$gdc_cases.project.primary_site) , pch = 19, cex = 1.5,
     xlab = 'PC5', ylab = 'PC6',
     main = 'PC (gene level) : log2(RPKM + 0.5)')

par(font.lab = 2, cex.lab = 1.2, font.axis = 2, cex.axis = 1.2)
plot(expr.pca.rpkm.log2.tcga$v[, 7], expr.pca.rpkm.log2.tcga$v[, 8], col= as.factor(md_TCGA$gdc_cases.project.primary_site) , pch = 19, cex = 1.5,
     xlab = 'PC7', ylab = 'PC8',
     main = 'PC (gene level) : log2(RPKM + 0.5)')
legend('topleft', pch = 19, col=c(1:2),
       names(summary(as.factor(md_TCGA$gdc_cases.project.primary_site))))

```

## Adjust for tissue

Looking at the separation across PCs1-4 above, we remove the effects of these PCs from the expression data before proceeding. 


```{r 'Cancer-healthy-variability', cache=TRUE, warning=FALSE}

PC1 <- expr.pca.rpkm.log2.tcga$v[, 1]
PC2 <- expr.pca.rpkm.log2.tcga$v[, 2]
PC3 <- expr.pca.rpkm.log2.tcga$v[, 3]
PC4 <- expr.pca.rpkm.log2.tcga$v[, 4]

## function to normalize expression removing effects of dataset
normalize_kidney <- function(x){
  #run model
  lmfit <- lm(x ~  PC1 + PC2+ PC3 + PC4)
  #adjust data
  #note: for quantitative, you have to subtract person's age from the average age of the group; for binary traits, you don't have to worry about this
  out <- x - (coef(lmfit)["PC1"]*(PC1-mean(PC1))) - (coef(lmfit)["PC2"]*(PC2-mean(PC2))) - (coef(lmfit)["PC3"]*(PC3-mean(PC3))) - (coef(lmfit)["PC4"]*(PC4-mean(PC4))) 
  return(out)
}


df <- apply(rpkm_log2_tcga,1,normalize_kidney)
rpkm_normalized_tcga <- t(df)

expr.pca.normalized.tcga <- svd(rpkm_normalized_tcga - rowMeans(rpkm_normalized_tcga))

par(mfrow=c(2,2))
par(font.lab = 2, cex.lab = 1.2, font.axis = 2, cex.axis = 1.2)
plot(expr.pca.normalized.tcga$v[, 1], expr.pca.normalized.tcga$v[, 2], col= as.factor(md_TCGA$gdc_cases.project.primary_site) , pch = 19, cex = 1.5,
     xlab = 'PC1', ylab = 'PC2',
     main = 'PC (gene level) : log2(normzlized counts + 0.5)')

par(font.lab = 2, cex.lab = 1.2, font.axis = 2, cex.axis = 1.2)
plot(expr.pca.normalized.tcga$v[, 3], expr.pca.normalized.tcga$v[, 4], col= as.factor(md_TCGA$gdc_cases.project.primary_site) , pch = 19, cex = 1.5,
     xlab = 'PC3', ylab = 'PC4',
     main = 'PC (gene level) : log2(normzlized counts + 0.5)')


par(font.lab = 2, cex.lab = 1.2, font.axis = 2, cex.axis = 1.2)
plot(expr.pca.normalized.tcga$v[, 5], expr.pca.normalized.tcga$v[, 6], col= as.factor(md_TCGA$gdc_cases.project.primary_site) , pch = 19, cex = 1.5,
     xlab = 'PC5', ylab = 'PC6',
     main = 'PC (gene level) : log2(normzlized counts + 0.5)')


par(font.lab = 2, cex.lab = 1.2, font.axis = 2, cex.axis = 1.2)
plot(expr.pca.normalized.tcga$v[, 7], expr.pca.normalized.tcga$v[, 8], col= as.factor(md_TCGA$gdc_cases.project.primary_site) , pch = 19, cex = 1.5,
     xlab = 'PC7', ylab = 'PC8',
     main = 'PC (gene level) : log2(normzlized counts + 0.5)')
legend('bottomright', pch = 19, col=c(1:3),
       names(summary(as.factor(md_TCGA$gdc_cases.project.primary_site))))

#Plot Variance Explained
par(mfrow=c(1,1))
plot(expr.pca.normalized$d^2/sum(expr.pca.normalized$d^2), xlim = c(0, 15), type = "b", pch = 16, xlab = "principal components", 
    ylab = "variance explained")
    
```

## Variable Expression Analysis

With expression data that can be compared across studies, we can then calculate variance across genes within each dataset. 

```{r variable-genes-kidney, warning=FALSE}
df = as.data.frame(t(rpkm_normalized_tcga))

## expression from each dataset

# N=899
kidney_df <- df %>% 
  filter(md_TCGA$gdc_cases.project.primary_site == 'Kidney')

brain_df <- df %>% 
  filter(md_TCGA$gdc_cases.project.primary_site == 'Brain')

## is measure variance across each dataset?
kidney_vars <- colVars(as.matrix(kidney_df))
brain_vars <- colVars(as.matrix(brain_df))


## add genenames back in
names(kidney_vars)  <- names(brain_vars) <- colnames(df)
```


## Concordance across studies

Again, we look at concordance between teh studies. Here, we're comaring TCGA's GBM data to TCGA's kidney tumor data. Concordance between these studies is included along with the previous concordance estimates.

```{r variable-concordance-kidney, cache=TRUE, warning=FALSE}
## sort by variance
p.mod4.sort <- kidney_vars[order(kidney_vars,decreasing=TRUE)]
p.mod5.sort <- brain_vars[order(brain_vars,decreasing=TRUE)]

conc_kidney <- NULL
for(i in 1:length(p.mod2.sort)){
    conc_kidney[i] <- sum(names(p.mod5.sort)[1:i] %in% names(p.mod4.sort)[1:i])
}



## top 1000 genes
par(mfrow = c(1, 1), font.lab = 1.5, cex.lab = 1.2, font.axis = 1.5, cex.axis = 1.2)
plot(seq(1:1000), conc[1:1000], 
     type = 'l', las = 0,
     xlim = c(0, 1000),
     ylim = c(0, 1000),
     xlab = 'ordered genes in reference study',
     ylab = 'ordered genes in new study',
     main = 'Concordance')
for(k in 1:10){
    abline(v = k * 200, cex = 0.5, col = 'lightgrey')
    abline(h = k * 200, cex = 0.5, col = 'lightgrey')
}
abline(coef=c(0,1),col="grey48")
lines(seq(1:1000), conc[1:1000], col = bright[2], lwd = 3)
lines(seq(1:1000), conc_TCGA[1:1000], col = bright[5], lwd = 3)
lines(seq(1:1000), conc_TCGA2[1:1000], col = bright[8], lwd = 3)
lines(seq(1:1000), conc_kidney[1:1000], col = bright[6], lwd = 3)
legend('topleft', col = bright[c(2,5,8,6)], c("SRP027383_SRP044668", "SRP027383_TCGA","SRP044668_TCGA", "brain_kidney"),lty=1,lwd=5, bg="white",bty="n")

```

As expected, lower concordance is found between kidney and GBM data sets, suggesting that genes variable across GBM datasets are likely specific to GBM pathology. 


# Reproducibility {.unnumbered}

```{r reproducibility}
## Reproducibility information
Sys.time()
proc.time()
options(width = 120)
devtools::session_info()
```


# References {.unnumbered}

The analyses were made possible thanks to:

* R `r citep(bib[['R']])`
* `r Biocpkg('BiocStyle')` `r citep(bib[['BiocStyle']])`
* `r Biocpkg('clusterProfiler')` `r citep(bib[['clusterProfiler']])`
* `r CRANpkg('devtools')` `r citep(bib[['devtools']])`
* `r CRANpkg('knitcitations')` `r citep(bib[['knitcitations']])`
* `r CRANpkg('knitr')` `r citep(bib[['knitr']])`
* `r Biocpkg('limma')` `r citep(bib[[c('limma', 'voom')]])`
* `r Biocpkg('recount')` `r citep(bib[[c('recount', 'recountWorkflow')]])`
* `r CRANpkg('rmarkdown')` `r citep(bib[['rmarkdown']])`


Full [bibliography file](example_multistudy.bib).

```{r bibliography, results='asis', echo=FALSE, warning = FALSE, message = FALSE}
## Print bibliography
bibliography()
```
