---
title: "Using recount_brain to explore relationships with post-mortem interval"
author:
- name: Leonardo Collado-Torres
  affiliation:
  - Lieber Institute for Brain Development, Johns Hopkins Medical Campus, Baltimore, MD, 21205, USA
  - Center for Computational Biology, Johns Hopkins University, Baltimore, MD, 21205, USA
  email: leo.collado@libd.org
output:
  BiocStyle::pdf_document
abstract: |
  This is an example of how you can use recount_brain to check if some results from another study are replicated in brain samples from the SRA present in recount2. This example also shows how you can quantify the percent of mitochondrial transcription and options for comparing two sets of SRA studies.
editor_options: 
  chunk_output_type: inline
---

```{r 'setup', echo = FALSE, warning = FALSE, message = FALSE}
## Bib setup
library('knitcitations')
library('BiocStyle')

## Load knitcitations with a clean bibliography
cleanbib()
cite_options(hyperlink = 'to.doc', citation_format = 'text', style = 'html')

## Write bibliography information
bib <- c(
    R = citation(),
    BiocStyle = citation('BiocStyle'),
    devtools = citation('devtools'),
    ggplot2 = citation('ggplot2'),
    knitcitations = citation('knitcitations'),
    knitr = citation('knitr')[3],
	recount = citation('recount')[1],
    recountWorkflow = citation('recount')[2],
    phenopredict = citation('recount')[3],
    rmarkdown = citation('rmarkdown'),
    original = bib_metadata('10.1038/s41467-017-02772-x'),
    jaffelab = citation('jaffelab')
)

write.bibtex(bib, file = 'example_PMI.bib')
```

<a href="https://jhubiostatistics.shinyapps.io/recount/"><img src="https://raw.githubusercontent.com/LieberInstitute/recount-brain/master/recount_brain.png" align="center"></a>

# Introduction

This document is an example of how you can use `recount_brain`. In a recent paper by Ferreira _et al._ called "The effects of death and post-mortem cold ischemia on human tissue transcriptomes" the authors explore associations between expression and post-mortem interval (PMI) `r citep(bib[['original']])`. We asked ourselves if PMI was recorded in any brain samples in `recount2` `r citep(bib[['recount']])` and got started using `recount_brain`.  Here we will check the expression trends of some of the genes highlighted by Ferreira _et al._ in [Figure 2d](https://www.nature.com/articles/s41467-017-02772-x/figures/2) and the association between PMI and percent of reads aligning to the mitochondrial chromosome from [Figure 4b and 4c](https://www.nature.com/articles/s41467-017-02772-x/figures/4). We will use the gene count data as well as the BigWig files provided by `recount2` `r citep(bib[['recount']])` and combine this information with the sample metadata from `recount_brain`.

This example shows how you can use the full `recount_brain` table to identify samples/studies of interest and the diversity of expression data available in `recount2`. This can be useful prior to publication if you want to check if your results are replicated in other data sets, for post-publication meta-analyses and for education purposes.


# Identify samples and studies of interest


Just like in the [example with data from SRP027373](../example_SRP027373/example_SRP027373.html), we will e will be using many functions from the `r Biocpkg('recount')` package, lets load it first^[If you are a first time `recount` user, we recommend first reading the package vignette at [bioconductor.org/packages/recount](http://bioconductor.org/packages/recount).].

```{r 'loadrecount', message = FALSE}
## Load the package
library('recount')
```

## Download `recount_brain`

Now that we have loaded `r Biocpkg('recount')`, lets obtain the full `recount_brain` table. This can be done via the `add_metadata()` function when we don't specify a `rse` argument.

```{r 'getrecountbrain'}
recount_brain <- add_metadata(source = 'recount_brain_v1')
dim(recount_brain)
```

## Explore `recount_brain` data

The full table includes samples not present in `recount2`, so lets drop those. 

```{r 'dropabsent'}
## Some studies included non-human Illumina RNA-seq data,
## which by design are absent in recount2
recount_brain <- recount_brain[recount_brain$present_in_recount, ]
dim(recount_brain)
```
Next, we can find which variables have information about post-mortem interval (PMI).

```{r 'pmivars'}
colnames(recount_brain)[grep('pmi', colnames(recount_brain))]
```

We can see that information about PMI is stored in two variables: the PMI value and the PMI unit. Lets keep only the samples where both of them are observed: most the samples with PMI values have a PMI unit.

```{r 'haspmi'}
with(recount_brain, addmargins(table(
    'PMI observed' = !is.na(pmi), 
    'Has PMI units' = !is.na(pmi_units)
    ))
)
recount_brain <- recount_brain[with(recount_brain,
    !is.na(pmi) & !is.na(pmi_units)), ]
dim(recount_brain)
```

Now that we have selected our samples of interest, lets check how many different studies these samples come from. As we can see below it's `r length(unique(recount_brain$sra_study_s))` studies with a range of `r min(table(recount_brain$sra_study_s))` to `r max(table(recount_brain$sra_study_s))` samples in each study.

```{r 'pmistudies'}
length(unique(recount_brain$sra_study_s))
sort(table(recount_brain$sra_study_s))
```

We can also explore the range of PMI values and units. Actually, all of the PMI values are measured in hours as we can see below.

```{r 'summaryunits'}
table(recount_brain$pmi_units)
summary(recount_brain$pmi)
```

In Figure \@ref(fig:pmidist) we plot the distribution of PMI in hours across all the samples of interest from the `r length(unique(recount_brain$sra_study_s))` studies. There are a few peaks which could be due to some tissues having less diverse PMI values.

```{r 'pmidist', fig.cap = 'Distribution of PMI in hours for all SRA studies with PMI information present in recount\\_brain.', message = FALSE}
library('ggplot2')

## Used code from
## http://www.cookbook-r.com/Graphs/Plotting_distributions_(ggplot2)/
ggplot(recount_brain, aes(x = pmi)) +
    geom_histogram(aes(y=..density..),
        colour="black", fill="white") +
    geom_density(alpha=.2, fill="#FF6666") + xlab('PMI in hours') +
    theme_grey(base_size = 18)
```

Figure \@ref(fig:pmistudy) shows the distribution of PMI in hours for each of the `r length(unique(recount_brain$sra_study_s))`. Two of them stand out as they have no variation in PMI: studies [SRP067645](https://trace.ncbi.nlm.nih.gov/Traces/sra/?study=SRP067645) and [SRP055440](https://trace.ncbi.nlm.nih.gov/Traces/sra/?study=SRP055440).

```{r 'pmistudy', fig.cap = 'Distribution of PMI in hours by study.'}
ggplot(recount_brain, aes(y = pmi, x = sra_study_s)) +
    geom_boxplot() + coord_flip() + theme_bw(base_size = 18) +
    ylab('PMI in hours') + xlab('SRA Study ID')
```

```{r 'dropnopmivar'}
recount_brain <- subset(recount_brain,
    !sra_study_s %in% c('SRP067645', 'SRP055440'))
## Final dimentions
dim(recount_brain)
```
We can now remake Figure \@ref(fig:pmidist) having dropped the two studies with no PMI variability as shown in Figure \@ref(fig:pmidistsub). There's still a peak at low PMI values, most likely from the 8 samples from study [SRP056604](https://trace.ncbi.nlm.nih.gov/Traces/sra/?study=SRP056604).


```{r 'pmidistsub', fig.cap = 'Distribution of PMI in hours for all SRA studies with PMI information present in recount\\_brain and that have PMI variability.', message = FALSE}
ggplot(recount_brain, aes(x = pmi)) +
    geom_histogram(aes(y=..density..),
        colour="black", fill="white") +
    geom_density(alpha=.2, fill="#FF6666") + xlab('PMI in hours') +
    theme_grey(base_size = 18)
```

Note that some of these studies use control samples and others focus on specific diseases. Figure \@ref(fig:pmidisease) shows the distribution of PMI by disease status where we see no significant difference. Though it could play a role in it's relationship with expression.

```{r 'pmidisease', fig.cap = 'Distribution of PMI in hours by disease status.'}
## What is the disease status of our current samples?
table(recount_brain$disease_status, useNA = 'ifany')

ggplot(recount_brain, aes(y = pmi, x = disease_status)) +
    geom_boxplot() + coord_flip() + theme_bw(base_size = 18) +
    ylab('PMI in hours') + xlab('Sample type')

## Is there a significant difference in PMI by disease status?
with(recount_brain, t.test(pmi ~ disease_status))
```


Awesome, we are now ready to get expression data!


# Gene expression

In [Figure 2d](https://www.nature.com/articles/s41467-017-02772-x/figures/2) Ferreira _et al._ `r citep(bib[['original']])` show the changes in gene expression for different tissues across several PMI intervals. We can download the gene count information from `recount2` using the `download_study()` function as shown next.

```{r 'getgene'}
## Our studies of interest
studies <- unique(recount_brain$sra_study_s)

## Download the files if they are missing
if(any(!file.exists(file.path(studies, 'rse_gene.Rdata')))) {
    sapply(studies, download_study)
}
```

Now that we have the gene RangedSummarizedExperiment (RSE) objects in our computer, we can proceed to load them. Since each file has a `rse_gene` object we have to be careful when loading these files. One way to handle them is to load them in a loop function such as a `lapply()` and `return()` them to create a list of RSE objects. Since the sample metadata variables and gene information is the same for each of these objects, we can combine them into a single object using `cbind()`.

```{r 'loadgene'}
## Load each rse_gene object
rse_gene <- lapply(studies, function(s) {
    message(paste(Sys.time(), 'processing study', s))
    load(file.path(s, 'rse_gene.Rdata'), verbose = TRUE)
    return(rse_gene)
})
## Combine them into a single one
rse_gene <- do.call(cbind, rse_gene)
dim(rse_gene)
```
Looks like we have one extra sample than what we were working with, so lets drop it. Once that's done we have our merged RSE object with the data from all `r length(studies)` studies.

```{r 'mergedrse'}
## Keep only the samples we were working with
rse_gene <- rse_gene[, colnames(rse_gene) %in% recount_brain$run_s]

## Check that we are all ok
stopifnot(ncol(rse_gene) == nrow(recount_brain))

## Merge rse_gene object
rse_gene
```


## Add `recount_brain` sample metadata

Now that we have the necessary gene information we can combine it with the sample metadata from `recount_brain`. We could either do this manually or simply use the `recount::add_metadata()` function.

```{r 'addrecountbrain'}
## Add recount_brain metadata
dim(colData(rse_gene))
rse_gene <- add_metadata(rse = rse_gene, source = 'recount_brain_v1')
dim(colData(rse_gene))
```

## PMI intervals

In [Figure 2d](https://www.nature.com/articles/s41467-017-02772-x/figures/2) Ferreira _et al._ `r citep(bib[['original']])` use 5 sets of PMI intervals. We can reconstruct them with the `cut()` function and use the `right = FALSE` argument to make sure that the intervals match theirs: [left inclusive, right exclusive). In our case we don't have any PMIs in the [0, 1) hour interval so we won't be using it.

```{r 'pmilvls'}
## Compute PMI intervals
colData(rse_gene)$pmilvl <- cut(rse_gene$pmi,
    c(0, 1, 4, 6, 15, max(rse_gene$pmi) + 1), right = FALSE)

## Check that the intervals match those from Figure 2d
table(colData(rse_gene)$pmilvl)

## Rename to I1, I2, ..., I5
levels(colData(rse_gene)$pmilvl) <- paste0('I', 1:5)

## Drop empty interval
colData(rse_gene)$pmilvl <- droplevels(colData(rse_gene)$pmilvl)
```

## Normalize expression


In [Figure 2d](https://www.nature.com/articles/s41467-017-02772-x/figures/2) Ferreira _et al._ `r citep(bib[['original']])` use $\log_2($ RPKM + 0.5 $)$. We can compute RPKM values using the `getRPKM()` function from `r Biocpkg('recount')`. 

```{r 'computerpkm'}
rpkm <- getRPKM(scale_counts(rse_gene))
```


In their case, all the expression data came from controls and from the same project. As a simple check, we can adjust the normalized expression for the disease status and project. We can do this with the `cleaningY()` function from `r Githubpkg('LieberInstitute/jaffelab')` just like we did in the [example with data from SRP027373](../example_SRP027373/example_SRP027373.html). The disease status and study might have stronger effects than what this approach can remove, but it can help us explore the data.


```{r 'get_jaffelab', message = FALSE}
library('jaffelab')
## You can install it with
# devtools::install_github('LieberInstitute/jaffelab')

mod <- with(colData(rse_gene),
    model.matrix(~ pmi + disease_status + project))
cleaned <- cleaningY(log2(rpkm + 0.5), mod, P = 2)
```

## Visualize example genes

In order to visualize the same genes as in [Figure 2d](https://www.nature.com/articles/s41467-017-02772-x/figures/2) `r citep(bib[['original']])` we need to first find these genes in the `recount2` data. This gets a little bit tricky because the symbol information is stored in a `CharacterList` object.

```{r 'find_genes'}
## Short function for finding the genes
find_gene <- function(symb) {
    ## Note that rowRanges(rse_gene)$symbol is a CharacterList object
    finding_gene <- rowRanges(rse_gene)$symbol == symb
    i <- which(any(finding_gene))
    return(i)
}

## Find them! ^_^
gene_i <- sapply(c('RNASE2', 'EGR3', 'HBA1', 'CXCL2'), find_gene)
gene_i
```

We finally have all the information we need:

* the gene expression data for the selected studies,
* the PMI interval in hours data,
* the gene IDs that match those used by Ferreira _et al._

Lets make a function that can plot either the normalized gene expression as in the original work, or the normalize gene expression after removing the effects of disease status and study. We might also be interested in sub setting by study to ask if the global trend could have been seen before and whether it would have been consistent. Let's also add the p-value for the linear regression between the PMI intervals and gene expression.

```{r 'gene_plot_code'}
## Our flexible plotting function.
## Mainly requires a gene symbol.
plot_gene <- function(symb, type = 'cleaned', study) {
    ## Keep the jitter reproducible
    set.seed(20180303)
    
    ## Find the gene based on the symbol
    i <- gene_i[grep(symb, names(gene_i))]
    
    ## Define what type of expression we are looking at
    if(type == 'cleaned') {
        y <- cleaned[i, ]
        ylab <- 'Norm Expr: disease status and project removed'
    } else if (type == 'norm') {
        y <- log2(rpkm[i, ] + 0.5) 
        ylab <- 'Normalized Expression'
    }
    x <- colData(rse_gene)$pmilvl
    ylim <- abs(range(y)) * c(0.95, 1.05) * sign(range(y))
    
    ## Subset to a given study
    if(!missing(study)) {
        y <- y[rse_gene$project == study]
        x <- x[rse_gene$project == study]
    }
    
    ## Plot components
    f <- lm(y ~ as.integer(x))
    title <- paste(symb, 'p-value', signif(summary(f)$coef[2, 4], 3))
    if(!missing(study)) title <- paste(title, 'study', study)
    
    ## Make the plot ^_^
    plot(y ~ x, xlab = 'PMI interval (h)', ylab = ylab, outline = FALSE,
        ylim = ylim, main = title)
    ## Colors from color blind friendly palette at
    ## http://www.cookbook-r.com/Graphs/Colors_(ggplot2)/
    points(y ~ jitter(as.integer(x), 0.6),
        bg = c("#009E73", "#0072B2", "#E69F00", "#D55E00")[
            as.integer(x)], pch = 21)
    abline(f, lwd = 3, col = "#CC79A7")
}
```

In Figures \@ref(fig:keygenes1), \@ref(fig:keygenes2), \@ref(fig:keygenes3) and \@ref(fig:keygenes4) we check if there's a relationship in the gene expression of _RNASE2_, _EGR3_, _HBA1_ and _CXCL2_ with the 4 PMI intervals (in hours) that are present in our data. The directions of the trends match those in [Figure 2d](https://www.nature.com/articles/s41467-017-02772-x/figures/2) of the original study across the same 4 PMI intervals. In particular, _CXCL2_ initially has a higher expression in artery (tibial) for I1, but it tends to increase across intervals I2 to I4 similar to what we observe in Figure \@ref(fig:keygenes4). In Figure \@ref(fig:keygenes1) we see a significant decrease (p < 0.05) in _RNASE2_ expression as the PMI interval increases.

```{r 'keygenes', fig.cap = paste('Normalized expression adjusted for disease status and project across PMI intervals for gene', c('RNASE2', 'EGR3', 'HBA1', 'CXCL2'))}
## Now make the plots
plot_gene('RNASE2')
plot_gene('EGR3')
plot_gene('HBA1')
plot_gene('CXCL2')
```

Ferreira _et al._ `r citep(bib[['original']])` found the same negative trend in _RNASE2_ expression through PMI interval progression for 13 tissues that they show in their [Figure 2d](https://www.nature.com/articles/s41467-017-02772-x/figures/2), but not in the brain. From Figure \@ref(fig:pmistudy) we know that not all the studies we are using span all PMI intervals. Using those studies that span at least 3 PMI intervals we explore the relationship between _RNASE2_ expression and PMI interval progression as shown in Figures \@ref(fig:rnase21), \@ref(fig:rnase22), \@ref(fig:rnase23) and \@ref(fig:rnase24). Three out of the four studies show a negative trend although it's not significant in any of them. SRP019762 shows an increase in _RNASE2_ expression but it also has no PMIs in the second interval and only two in the fifth interval.

```{r 'rnase2studies'}
## Plot RNASE2 by study
study_nint <- with(colData(rse_gene),
    tapply(pmilvl, project, function(x) length(unique(x))))
study_nint <- names(study_nint)[study_nint >= 3]
```


```{r 'rnase2', fig.cap = paste('Normalized RNASE2 expression for study', study_nint)}
for(s in study_nint) plot_gene('RNASE2', type = 'norm', study = s)
```

The variability across studies shown in Figures \@ref(fig:rnase21), \@ref(fig:rnase22), \@ref(fig:rnase23) and \@ref(fig:rnase24) as well as the variability in brain regions these studies represent could explain why Ferreira _et al._ didn't see a significant relationship between _RNASE2_ expression and PMI interval in their study.

```{r 'study_nint_subtissue'}
with(colData(rse_gene[, rse_gene$project %in% study_nint]),
    table(tissue_site_1, project, useNA = 'ifany'))
with(colData(rse_gene[, rse_gene$project %in% study_nint]),
    table(tissue_site_3, project, useNA = 'ifany'))
```


# Mitochondrial expression and PMI

In [Figure 4b and 4c](https://www.nature.com/articles/s41467-017-02772-x/figures/4) Ferreira _et al._ `r citep(bib[['original']])` show how the percent of mitochondrial transcription increases as PMI increases for liver, but decreases for the ovary tissue. In their [Supplementary Figure 17](https://static-content.springer.com/esm/art%3A10.1038%2Fs41467-017-02772-x/MediaObjects/41467_2017_2772_MOESM1_ESM.pdf) they show that in the brain (cerebellum and cortex) mitochondrial expression and PMI as positively correlated.

One of the advantages of `recount2` is that it provides expression data summarized at different feature types (genes, exons, junctions) and that it provides base-pair resolution coverage BigWig files `r citep(bib[[c('recount', 'recountWorkflow')]])`. Meaning that we can quantify the levels of mitochondrial expression in the same samples and studies we have been working on.

## Quantify chrM expression

As explained in detail in the recount workflow paper `r citep(bib[['recountWorkflow']])`, one of the variables included in `recount2` is the _AUC_ (area under coverage) which is the sum of the base-pair coverage across all the genome. With the `coverage_matrix()` function we can compute the _AUC_ for a given set of region coordinates. So using the [chromosome size data for hg38](https://github.com/nellore/runs/blob/master/gtex/hg38.sizes) we can define a single region spanning the whole mitochondrial chromosome. Then we can quantify the `chrM` _AUC_ for all of our `r ncol(rse_gene)` samples. This will take about 10 to 25 minutes depending on your internet connection. 

```{r 'compute_chrM_auc', cache = TRUE}
## Got chrM size from
## https://github.com/nellore/runs/blob/master/gtex/hg38.sizes
mt <- GRanges('chrM', IRanges(1, 16569))

## Compute the chrM auc
## Turn scaling off by default so we can then use
## colData(rse_gene)$auc
rse_chrM <- lapply(studies, function(study) {
    message(paste(Sys.time(), 'processing study', study))
    coverage_matrix(project = study, chr = 'chrM', regions = mt,
        chrlen = 16569, scale = FALSE,
        verboseLoad = FALSE, verbose = FALSE)
})

## Combine the list of RSE objects into a single RSE
rse_chrM <- do.call(cbind, rse_chrM)

## Match the order of the samples we have in rse_gene
rse_chrM <- rse_chrM[, match(rse_gene$run, rse_chrM$run)]
```


Note that if we had the BigWig files in our computer, we could use the `coverage_matrix_bwtool()` function from the `r Githubpkg('LieberInstitute/recount.bwtool')` package `r citep(bib[['phenopredict']])`. Although 25 minutes is not a terrible time to wait. If you don't want to wait, check the [example_results.Rdata](example_results.Rdata) file that we create later on and that contains this information.


## Percent of chrM expression

Now we have all the pieces we need for calculating the percent of mitochondrial expression. Lets save that information in our `rse_gene` object.

```{r 'chrM_percent'}
## chrM auc and percent saved in our sample metadata
colData(rse_gene)$chrM_auc <- assays(rse_chrM)$counts[1, ]
colData(rse_gene)$chrM_percent <- colData(rse_gene)$chrM_auc /
    colData(rse_gene)$auc * 100
```


## Percent of reads in genes

The gene data in `recount2` is actually the _AUC_ for the exonic sequence of each gene. So we can use this information to compute the percent of expression in genes (based on Gencode v25). In order to compare it against the mitochondrial expression, we exclude the mitochondrial genes.

```{r 'gene_percent'}
## Exclude chrM genes
colData(rse_gene)$gene_auc <- colSums(assays(rse_gene)$counts[
    as.logical(seqnames(rowRanges(rse_gene)) != 'chrM'), ])
colData(rse_gene)$gene_percent <- colData(rse_gene)$gene_auc /
    colData(rse_gene)$auc * 100

## Check that none go above 100
with(colData(rse_gene), summary(chrM_percent + gene_percent))
```


## PMI and chrM expression

Now that we wave all the information we need, we can visualize the relationship between percent of mitochondrial transcription and PMI in hours as shown in Figure \@ref(fig:mtpmi). The trend is positive just like Ferreira _et al._ found in their [Supplementary Figure 17](https://static-content.springer.com/esm/art%3A10.1038%2Fs41467-017-02772-x/MediaObjects/41467_2017_2772_MOESM1_ESM.pdf) for brain cortex and brain cerebellum. However, compared to their [Figure 4b and 4c](https://www.nature.com/articles/s41467-017-02772-x/figures/4), we observe a higher range of mitochondrial transcription. The trend we observe is significantly different from 0.

```{r 'mtpmi', fig.cap = 'PMI in hours is positively associated with mitochondrial transcription.'}
## For simplifying the calls later on
d <- as.data.frame(colData(rse_gene))

## Save plot object for the next plot
g <- ggplot(d, aes(y = chrM_percent, x = pmi)) + geom_point() +
    geom_smooth(method = 'lm') + xlab('PMI in hours') +
    ylab('MT%') + theme_light(base_size = 18)
g

## Is the association significant? yes
summary(with(d, lm(chrM_percent ~ pmi)))
```

Figure \@ref(fig:mtpmidisease) breaks up samples by disease status. The trend has the same direction although the intercepts do change.

```{r 'mtpmidisease', fig.cap = 'PMI in hours is positively associated with mitochondrial transcription regardless of disease status.', fig.wide = TRUE}
g + facet_grid(~ disease_status)
```

In Figure \@ref(fig:mtpmistudy) we observe that some studies have a positive and others a negative relationship between PMI in hours and the percent of mitochondrial transcription. Although the standard errors are wide, for study SRP058181 we see that there is an opposite relationship between cases and controls, while it remains negative by disease status for SRP051844.

```{r 'mtpmistudy', fig.cap = 'PMI in hours and mitochondrial transcription by study for studies with PMI values across at least 3 intervals.', fig.wide = TRUE}
## Make the projects into a factor so that we can use
## scale_colour_hue(drop = FALSE) when plotting a subset 
## of the studies and the colors will stay consistent when
## using the full data
d$project <- as.factor(d$project)

## Use only the studies that have PMI values at least in 3 PMI intervals
ggplot(subset(d, project %in% study_nint),
    aes(y = chrM_percent, x = pmi, colour = project)) +
    geom_point() + geom_smooth(method = 'lm') +
    xlab('PMI in hours') + ylab('MT%') +
    theme_light(base_size = 18) +
    scale_colour_hue(l = 40, drop = FALSE, name = 'study') +
    facet_grid(~ disease_status)
```

## PMI and gene expression


Just like we explored the association between PMI in hours and mitochondrial expression in Figure \@ref(fig:mtpmi), in Figure \@ref(fig:genepmi) we check the relationship between PMI in hours and the percent of transcription that overlaps known genes (from Gencode v25) excluding the mitochondrial genes. The global trend is also slightly positive.

```{r 'genepmi', fig.cap = 'PMI in hours is positively associated with global gene transcription excluding mitochondrial genes.'}
g2 <- ggplot(d, aes(y = gene_percent, x = pmi)) + geom_point() +
    geom_smooth(method = 'lm') + xlab('PMI in hours') +
    ylab('Gene % (excluding MT)') + theme_light(base_size = 18)
g2
```

Unlike Figure \@ref(fig:mtpmidisease), in Figure \@ref(fig:genepmidisease) we do see different trends by disease status. However, the error bands are wide and we can see in a linear regression that neither the disease status term nor the interaction between disease status and PMI in hours are significant. 

```{r 'genepmidisease', fig.wide = TRUE, fig.cap = 'PMI in hours is positively associated with global gene transcription excluding mitochondrial genes for control samples and has no association for the samples wiht a disease phenotype, although they are not significantly different.'}
g2 + facet_grid(~ disease_status)

## Use a linear regresion and check if the
## disease_status coefficient is significantly different from 0
summary(with(d, lm(gene_percent ~ pmi * disease_status)))

## Could also do the following if we assume parallel lines
## Results in p-value > 0.05 for the disease status term
# summary(with(d, lm(gene_percent ~ pmi + disease_status)))
```
Figure \@ref(fig:genepmistudy) is the equivalent of Figure \@ref(fig:mtpmistudy) where we again see variability between studies with wide standard errors.


```{r 'genepmistudy', fig.wide = TRUE, fig.cap = 'PMI in hours and global gene transcription excluding mitochondrial genes by study for studies with PMI values across at least 3 intervals.'}
ggplot(subset(d, project %in% study_nint),
    aes(y = gene_percent, x = pmi, colour = project)) +
    geom_point() + geom_smooth(method = 'lm') +
    xlab('PMI in hours') + ylab('Gene % (excluding MT)') +
    theme_light(base_size = 18) +
    scale_colour_hue(l = 40, drop = FALSE, name = 'study') +
    facet_grid(~ disease_status)
```

## chrM vs gene expression

Lets now compare the mitochondrial transcription vs the global gene transcription (excluding mitochondrial genes) percents directly. We expect that high levels of mitochondrial transcription will lead to lower levers of global gene expression. Figure \@ref(fig:mtgene0) shows that the expected negative association is true, but not for all samples. There is either an elbow curve, or two sets of relationships: one where it's first increasing, and another one that is decreasing as we expected.

```{r 'mtgene0', fig.cap = 'Mitochondrial expression versus global gene expression excluding mitochondrial genes has an elbow shape or two types of relationships (one increasing, one decreasing).'}
ggplot(d, aes(y = gene_percent, x = chrM_percent)) +
    geom_point() + ylab('Gene % (excluding MT)') +
    xlab('MT%') + theme_light(base_size = 18) +
    geom_smooth(method = 'loess')
```


Figure \@ref(fig:mtgene) shows that PMI interval doesn't seem to play a role in discriminating between the two groups of samples.

```{r 'mtgene', fig.cap = 'Mitochondrial expression versus global gene expression excluding mitochondrial genes is not influenced by PMI interval.'}
ggplot(d, aes(y = gene_percent, x = chrM_percent, colour = pmilvl)) +
    geom_point() + ylab('Gene % (excluding MT)') +
    xlab('MT%') + theme_light(base_size = 18) +
    scale_colour_brewer(palette = 'PuOr', name = 'PMI interval')
```

In Figure \@ref(fig:mtgene2) we separate the samples by disease status and study. Most of the samples from a disease phenotype have the expected negative relationship. However, the disease status doesn't really discriminate between the samples in projects SRP056604 and SRP048683. 

```{r 'mtgene2', fig.wide = TRUE, fig.cap = 'Mitochondrial expression versus global gene expression excluding mitochondrial genes depends on study rather than disease status.'}
ggplot(d, aes(y = gene_percent, x = chrM_percent, colour = project)) +
    geom_point() + facet_grid(~disease_status) +
    ylab('Gene % (excluding MT)') + xlab('MT%') +
    theme_light(base_size = 18) +
    scale_colour_hue(l = 40, name = 'study')
```

In Figure \@ref(fig:mtgene2) we observe that the studies do help differentiate the samples. If we group the studies in the following two sets:

* set1: SRP017933, SRP032539, SRP032540, SRP048683, SRP056604
* set2: ERP001304, SRP051844, SRP007483, SRP058181, SRP019762

then the samples are perfectly grouped with the two types of relationships between global gene and mitochondrial expression as shown in Figure \@ref(fig:mtgene3).


```{r 'mtgene3', fig.cap = 'Mitochondrial expression versus global gene expression excluding mitochondrial genes depends on study.'}
## Group studies in two sets
set1 <- c('SRP017933', 'SRP032539', 'SRP032540', 'SRP048683', 'SRP056604')
colData(rse_gene)$study_set <- ifelse(
    colData(rse_gene)$project %in% set1, 'set1', 'set2')

## Remake the d data.frame
d <- as.data.frame(colData(rse_gene))
d$project <- as.factor(d$project)

## Now make the plot
ggplot(d, aes(y = gene_percent, x = chrM_percent,
    colour = study_set)) + geom_point() +
    ylab('Gene % (excluding MT)') + xlab('MT%') +
    theme_light(base_size = 18) +
    geom_smooth(method = 'lm') +
    scale_colour_hue(l = 30, name = 'Study set')
```

Lets revise the relationship between PMI and mitochondrial expression from Figure \@ref(fig:mtpmi) by separating the studies as shown in Figure \@ref(fig:mtgene4). For the studies from the first set we see a range of mitochondrial transcription much closer than the one shown by Ferreira _et al._ `r citep(bib[['original']])` for liver and ovary tissues in their [Figure 4b and 4c](https://www.nature.com/articles/s41467-017-02772-x/figures/4). The relationship is negative, contrary to their findings in [Supplementary Figure 17](https://static-content.springer.com/esm/art%3A10.1038%2Fs41467-017-02772-x/MediaObjects/41467_2017_2772_MOESM1_ESM.pdf) for the brain cerebellum and cortex. Studies from the second set are more in line with our previous observations from Figure \@ref(fig:mtpmi). Interestingly, there is no significant interaction between PMI in hours and the study sets for explaining the percent of mitochondrial transcription. That is, the main difference is in the intercepts. 

```{r 'mtgene4', fig.wide = TRUE, fig.cap = 'PMI in hours is positively associated with mitochondrial transcription for studies from set2.'}
ggplot(d, aes(y = chrM_percent, x = pmi)) + geom_point() +
    geom_smooth(method = 'lm') + xlab('PMI in hours') +
    ylab('MT%') + theme_light(base_size = 18) +
    facet_grid(~ study_set)

## There is a significant difference by study set
summary(with(d, lm(chrM_percent ~ pmi + study_set)))

## The intercepts are significantly different, 
## but the interaction term isn't
summary(with(d, lm(chrM_percent ~ pmi * study_set)))
```
Note that the PMI term, that is the association between PMI measured in hours and mitochondrial expression, is not significant. This is in contrast to our initial observation in Figure \@ref(fig:mtpmi).

## Differences between sets of studies

These two sets of studies might have some properties that differentiate them. In this section we explore in many different ways the differences between the study sets. 

As we saw in Figure \@ref(fig:mtgene2), the first set is mostly made up of control samples. Also, the PMI intervals don't seem to play a role just like we observed in Figure \@ref(fig:mtgene).

```{r 'check1'}
with(colData(rse_gene), table(disease_status, study_set, useNA = 'ifany'))
with(colData(rse_gene), table(pmilvl, study_set, useNA = 'ifany'))
```


### Biological

Next we explore some biological covariates. Samples from the second set are all from the cerebral cortex although some samples from the first set are also from the cerebral cortex. For other variables such as the Brodmann area we don't have information for all the samples.

```{r 'check2'}
with(colData(rse_gene), table(tissue_site_1, study_set, useNA = 'ifany'))
with(colData(rse_gene), table(tissue_site_2, study_set, useNA = 'ifany'))
with(colData(rse_gene), table(tissue_site_3, study_set, useNA = 'ifany'))
with(colData(rse_gene), table(brodmann_area, study_set, useNA = 'ifany'))
with(colData(rse_gene), table(hemisphere, study_set, useNA = 'ifany'))
```

### Demographics

By demographics, we don't see a difference by sex, race (mostly missing) and age for the samples in the two study sets.

```{r 'check3'}
with(colData(rse_gene), table(sex, study_set, useNA = 'ifany'))
with(colData(rse_gene), table(race, study_set, useNA = 'ifany'))
with(colData(rse_gene), tapply(age, study_set, summary))
with(colData(rse_gene), t.test(age ~ study_set))
```

### Quality

32 out of the 67 samples from the first set have RIN values with most of the samples in the second set being observed (167 / 185). There is a significant difference in mean RIN between the sets, with the first set having a lower mean RIN. When we use RIN in the graph comparing the percents of transcription in Figure \@ref(fig:check4), RIN doesn't seem to be as good at differentiating the samples.

```{r 'check4', fig.cap = 'Mitochondrial expression versus global gene expression excluding mitochondrial genes with RIN levels.'}
with(colData(rse_gene), tapply(rin, study_set, summary))
with(colData(rse_gene), t.test(rin ~ study_set))

ggplot(d, aes(y = gene_percent, x = chrM_percent,
    colour = rin)) + geom_point() +
    ylab('Gene % (excluding MT)') + xlab('MT%') +
    theme_light(base_size = 18) +
    scale_colour_gradientn(colours = c('darkred', 'orange'), name = 'RIN')
```


### Technical

We next compare technical covariates between the studies. Samples from the first and second set are mostly single end and paired end, respectively. However, there's some overlap.

```{r 'check5'}
with(colData(rse_gene), table(paired_end, study_set, useNA = 'ifany'))
```

Looking at library selection, brain bank (assuming that "NICHD Brain and Tissue Bank" and "NICHD Brain and Tissue Bank for Developmental Disorders" are the same ones), instrument, and date of submission we don't see any striking differences between the sets of studies.

```{r 'check6'}
## Check technical variables
with(colData(rse_gene), table(libraryselection_s, study_set, useNA = 'ifany'))
with(colData(rse_gene), table(brain_bank, study_set, useNA = 'ifany'))
with(colData(rse_gene), table(instrument_s, study_set, useNA = 'ifany'))
with(colData(rse_gene), 
     table(ss(biosample_submission_date, '-'), study_set, useNA = 'ifany'))
```


Studies from set 1 do have lower median number of reads and mapped reads, however there is no significant difference in the means of these two covariates by study set.

```{r 'check7'}
with(colData(rse_gene), tapply(reads_downloaded, study_set, summary))
with(colData(rse_gene), tapply(mapped_read_count, study_set, summary))

with(colData(rse_gene), t.test(reads_downloaded ~ study_set))
with(colData(rse_gene), t.test(mapped_read_count ~ study_set))
```

If we look at the overall amount of transcription using the _AUC_, we do see a difference between study sets in both the paired end and the single end samples as shown in Figure \@ref(fig:check8). However, the study group with the highest _AUC_ is not consistent. Note that the _AUC_ is roughly the number of mapped reads multiplied by the average read length.

```{r 'check8', fig.cap = 'Global expression by AUC between study sets and library protocols.'}
d$paired_end_text <- ifelse(d$paired_end, 'Paired end', 'Single end')
ggplot(d, aes(y = auc, x = study_set)) + geom_boxplot() +
    scale_y_log10() +
    facet_grid( ~ paired_end_text) +
    theme_light(base_size = 18) + ylab('AUC') + 
    xlab('Study set')
```


### Read length

In Figure \@ref(fig:check9) there is a difference in average read length between the studies^[For paired-end studies, this is the sum of the length of both paired reads.]. But in a closer inspection of the outliers in the second set, we see that there are several samples with shorter read lengths.

```{r 'check9', fig.cap = 'Average read length by study group.'}
ggplot(d, aes(y = avgspotlen_l, x = study_set)) + geom_boxplot() +
    theme_light(base_size = 18) + ylab('Average read length') + 
    xlab('Study set')

with(colData(rse_gene), table(avgspotlen_l[study_set == 'set2']))
```

Within the samples from the second set, read length doesn't seem to inform the relationship between mitochondrial and global gene expression as shown in Figure \@ref(fig:check10).


```{r 'check10', fig.cap = 'Mitochondrial and gene expression for samples from the second set of studies by average read length cut at 200 base-pairs.'}
ggplot(d[d$study_set == 'set2', ], aes(y = gene_percent, x = chrM_percent,
    colour = avgspotlen_l < 200)) + geom_point() +
    facet_grid(~study_set) + ylab('Gene % (excluding MT)') +
    xlab('MT%') + theme_light(base_size = 18) +
    scale_colour_brewer(palette = 'Set1', name = 'Avg Read\nLength < 200')
```


### From predictions

Using the SHARQ beta sample metadata we see that there are cortex samples in both sets, so again, no clear cut separation.

```{r 'check11'}
## Metadata from SHARQ beta
with(colData(rse_gene), table(sharq_beta_cell_type, study_set, useNA = 'ifany'))
```

We also have predicted phenotype data `r citep(bib[['phenopredict']])` which we can get using the `add_predictions()` function from `r Biocpkg('recount')`. None of the samples are predicted to be cell lines, and we again see that most samples are from males with 38 / 42 samples matching the reported and the predicted sex for set 1 and all of them matching for set 2.

```{r 'check12'}
## Get the predictions from phenopredict
rse_gene <- add_predictions(rse_gene)

## Check sample source, and sex
with(colData(rse_gene),
    table(predicted_samplesource, study_set, useNA = 'ifany'))
with(colData(rse_gene),
    table(predicted_sex, study_set, useNA = 'ifany'))
with(colData(rse_gene),
    addmargins(table(predicted_sex, sex, study_set, useNA = 'ifany')))
```

## Save results

We finally save our exploratory results in case we want to carry out more analyses with them later on.


```{r save_res}
## Save results
save(rse_gene, rse_chrM, file = 'example_results.Rdata')
```


# Conclusions

In this document we showed how you can identify samples and studies of interest using `recount_brain` to check if some results from another study are replicated in brain samples from the SRA present in `recount2`. We found that _RNASE2_ expression is negatively associated with PMI intervals and show that the variability across studies and brain regions could be why Ferreira _et al._ `r citep(bib[['original']])` didn't observe a significant association in their study. We observed a significant increase in mitochondrial transcription with increasing PMI measured in hours that matches the observations by Ferreira _et al._ for brain cortex and brain cerebellum. However, we identified two sets of studies that present different relationships between mitochondrial and global gene transcription (excluding mitochondrial genes). Once we considered the two of studies we found that there is no significant association between mitochondrial transcription and PMI measured in hours. We also showed how you can compare two sets of studies using sample metadata from `recount_brain`, [SHARQ beta](http://www.cs.cmu.edu/~ckingsf/sharq/about.html) information present in `recount2` and predicted sample metadata `r citep(bib[['phenopredict']])`.

# Reproducibility {.unnumbered}

```{r reproducibility}
## Reproducibility information
Sys.time()
proc.time()
options(width = 120)
devtools::session_info()
```


# References {.unnumbered}

The analyses were made possible thanks to:

* R `r citep(bib[['R']])`
* `r Biocpkg('BiocStyle')` `r citep(bib[['BiocStyle']])`
* `r CRANpkg('devtools')` `r citep(bib[['devtools']])`
* `r CRANpkg('ggplot2')` `r citep(bib[['ggplot2']])`
* `r Githubpkg('LieberInstitute/jaffelab')` `r citep(bib[['jaffelab']])`
* `r CRANpkg('knitcitations')` `r citep(bib[['knitcitations']])`
* `r CRANpkg('knitr')` `r citep(bib[['knitr']])`
* `r Biocpkg('recount')` `r citep(bib[[c('recount', 'recountWorkflow', 'phenopredict')]])`
* `r CRANpkg('rmarkdown')` `r citep(bib[['rmarkdown']])`

Full [bibliography file](example_PMI.bib).

```{r bibliography, results='asis', echo=FALSE, warning = FALSE, message = FALSE}
## Print bibliography
bibliography(style = 'html')
```
